src/plugins/irc-events/link.js

```diff
@@ -436,7 +444,7 @@ function fetch(uri, headers) {
 						type = contentType.split(/ *; */).shift();
 					}
 
-					resolve({data: buffer, type, size});
+					resolve({data: buffer, type, size, filename: filename});
 				});
```

Comment:
style nit: `resolve({data: buffer, type, size, filename});` would be better as it's consistent with its surroundings.

Reply:
ah lel didn't really realize that that works. Am not so fluent in JS :)